src/test/kotlin/com/depromeet/team3/product/service/gemini/GeminiProductExtractorTest.kt

```
@@ -0,0 +1,38 @@
+    lateinit var extractor: GeminiProductExtractor
+
+    @Test
+    @Timeout(value = 90, unit = TimeUnit.SECONDS)
+    fun `Gemini end-to-end 호출이 살아 있고 응답을 구조화해 돌려준다`() {
+        // 외부 URL 에 대한 Gemini 의 isProductPage 판정이 유동적이라 Product 반환과
+        // notProductPage 예외 모두 정적 fetch + structured output 경로의 정상 동작 신호로 본다.
+        val link = ProductLink.parse("https://www.apple.com/shop/buy-iphone/iphone-15")
+
+        try {
+            extractor.extract(link)
+        } catch (e: ProductExtractionException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)